test/views/RemoveUnderpaymentDetailsViewSpec.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import base.ViewBaseSpec
+import forms.underpayments.RemoveUnderpaymentDetailsFormProvider
+import messages.RemoveUnderpaymentReasonMessages
+import messages.underpayments.RemoveUnderpaymentDetailsMessages
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.data.Form
+import play.api.mvc.Call
+import play.twirl.api.Html
+import views.html.underpayments.RemoveUnderpaymentDetailsView
+
+class RemoveUnderpaymentDetailsViewSpec extends ViewBaseSpec {
+
+  private lazy val injectedView: RemoveUnderpaymentDetailsView = app.injector.instanceOf[RemoveUnderpaymentDetailsView]
+
+  val underpaymentType = "B00"
```

Comment:
this needs to be a sequence of all the underpayment types, and you should check it's fine for all of them